test/runnable/builtin.d

```diff
@@ -45,10 +45,6 @@ void test2()
     i = i ^^ 2;
     assert(i == 9);
 
-    int j = 2;
-    j = j ^^ 1;
```

Comment:
I think this fails with this PR due to some aggressive const folding.  Interestingly `j = j + 0` does not generate an error, but `j ^^ 1` does.  I would prefer to add this back into the test suite, but a decision needs to be made about how it *should* behave.

Reply:
The question "how far should constant folding be done" is a hopeless one. In this case, no error should have been given.

Also,
```
int j = j;
```
must not be an error, because the `j` on the right is looked up in the enclosing scope - it's not the same `j`.

Reply:
I just tested the following with the current implementation:

test.d
```D
module test;

int t;
```

main.d
```D
import test;

void main()
{
    int j = j;
}
```

Under the current implementation `int j = j` will not generate an error.  The implementation is quite precise.

Reply:
> In this case, no error should have been given.

I'll work on implementing a fix for `j = j ^^ 1`